pkg/apis/devconsole/v1alpha1/component_types_labels.go

```diff
@@ -0,0 +1,36 @@
+package v1alpha1
+
+// GetLabelName retrieves the label name. If not present,
+// infer the name by using the component name value.
+func (c *Component) GetLabelName() string {
+	name := c.Labels["app.kubernetes.io/name"]
```

Comment:
What if the key not present? Should we handle it? Same for below label access.

Reply:
If any of the labels aren't present then an empty string is returned. Except for the name and instance fields. These two fields will infer these values if no labels are present in the custom resource. 

In devconsole-operator if an empty string is retrieved for a label, then it is not added to the labels map and hence will not appear. 

Maybe an easier way to understand how it works is by looking at this PR: https://github.com/redhat-developer/devconsole-operator/pull/114 and try running the component_controller_test.go file - the tests add all the labels, but you can remove one and see the results